GliaWidgetsTests/SecureConversations/Welcome/SecureConversations.WelcomeViewModelSpec.swift

```diff
@@ -0,0 +1,214 @@
+                expectation.fulfill()
+            default: XCTFail()
+            }
+        }
+
+        viewModel.event(.closeTapped)
+
+        waitForExpectations(timeout: 1)
+        XCTAssertEqual(isCalled, true)
+    }
+}
+
+// Is attachment available
+extension SecureConversationsWelcomeViewModelTests {
+    func testIsAttachmentAvailable() {
+        var environment: SecureConversations.WelcomeViewModel.Environment = .mock
+        environment.fetchSiteConfigurations = { completion in
+            let site: CoreSdkClient.Site = try! .mock(
+                allowedFileSenders: .init(operator: true, visitor: true)
+            )
+
+            completion(.success(site))
+        }
+
+        viewModel = .init(environment: environment, availability: .mock)
+
+        XCTAssertTrue(viewModel.isAttachmentsAvailable)
+    }
+
+    func testIsAttachmentNotAvailable() {
+        var environment: SecureConversations.WelcomeViewModel.Environment = .mock
+        environment.fetchSiteConfigurations = { completion in
+            let site: CoreSdkClient.Site = try! .mock(
+                allowedFileSenders: .init(operator: true, visitor: false)
+            )
+
+            completion(.success(site))
+        }
+
+        viewModel = .init(environment: environment, availability: .mock)
+
+        XCTAssertFalse(viewModel.isAttachmentsAvailable)
+    }
+
+    func testIsAttachmentAvailableFailed() {
+        let expectation = expectation(description: "")
+
+        var environment: SecureConversations.WelcomeViewModel.Environment = .mock
+        environment.fetchSiteConfigurations = { completion in
+            // Because the attachment availability is checked on the init, without
+            // this async, the fetching is so quick that there's not enough time for
+            // it to happen after the delegate is set.
+            DispatchQueue.main.async {
+                completion(.failure(CoreSdkClient.GliaCoreError(reason: "")))
+            }
+        }
+
+        var isCalled = false
+        var messageAlertConfiguration: MessageAlertConfiguration?
+
+        viewModel = .init(environment: environment, availability: .mock)
+        viewModel.delegate = { event in
+            switch event {
+            case .showAlert(let configuration, _, _):
+                isCalled = true
+                messageAlertConfiguration = configuration
+                expectation.fulfill()
+            default: break
+            }
+        }
+
+        waitForExpectations(timeout: 1)
+        XCTAssertTrue(isCalled)
+        XCTAssertTrue(messageAlertConfiguration?.message == self.viewModel.environment.alertConfiguration.unexpectedError.message)
+    }
+
+}
+
+// Send message
+extension SecureConversationsWelcomeViewModelTests {
+    func testSuccessfulMessageSend() {
+        let expectation = expectation(description: "")
+        var isCalled = false
+        self.viewModel.environment.sendSecureMessage = { _, _, _, completion in
```

Comment:
In each test above you initialize viewModel, but here you don't, so changing viewModel in previous test might affect  it in following tests. Look like would be better to initialize viewModel in each test or in setup/tearDown method. 
What do you think?

Reply:
You are correct. This already bit me in one of the tests I am writing now. Will add a `setUp` override.